### PR TITLE
Drop Autotools support and release version 1.22.0

### PR DESCRIPTION
**Pull request by mgrabovsky**
The [current release procedure](https://github.com/abrt/retrace-server/wiki/Release-Checklist) seems to be broken, so I'm checking to see if this is what a new release should look like.

### CODE REVIEW

**Comment by michalfabik** [comment]
What was the problem with the current release procedure? Are you referring to the `CHANGELOG.md` that you removed from the instructions? I had an issue with that too when releasing FAF but I found I had to stage it before running `tito tag` so it ends up in the same commit with the release (i.e. the changes made by `tito`).

**Comment by mgrabovsky** [comment]
The changelog is one thing – there's no `CHANELOG.md` in retrace-server. There's only the `%changelog` section the spec file, which needs to be amended manually.

There are other issues, though, and I don't have answers to all the questions that pop up in my head.

1. When I ran `tito tag --use-version 1.22.0`, it only bumped the version in `retrace-serve.spec`. (Is this the correct behaviour? Shouldn't it be done in the `.spec.in` file? Why do we have the `.spec` under version control when `.spec.in` is the decisive one?)
2. The version in `gen-version` was not updated. (Is this file still in use now that we have Meson and Tito?)

Ernestas made some changes after the last release to incorporate Tito, the main one being 6338fb3, I think. The last release was months so we haven't had the chance to test the new process yet.

I believe these issues should be easy to fix by tweaking the Tito config, but we need to decide what the intended behaviours and outcomes are.

**Comment by mkutlak** [comment]
From https://github.com/abrt/retrace-server/pull/315:
> Autotools support is not nuked yet, but will be soon.

I guess it is time.

> When I ran `tito tag --use-version 1.22.0`, it only bumped the version in `retrace-serve.spec`. (Is this the correct behaviour? Shouldn't it be done in the `.spec.in` file? Why do we have the `.spec` under version control when `.spec.in` is the decisive one?)

`spec.in` wass used by `configure` to generate the spec file. After tito/meson was introduced the `spec.in` is no longer used.

> The version in `gen-version` was not updated. (Is this file still in use now that we have Meson and Tito?)

 Yes, `tito` is used to update version of release instead of `gen-version`.

File: retrace-server.spec.in [hunk 1]
```diff
@@ -198,6 +198,28 @@ exit 0
...
+- Drop python3-six dependency
+- Add build dependencies on gzip, lsof, tar and xz
+- Rewrite Dockerfile
+- Migrate to calling subprocess.run() in place of s.Popen() and s.call()
+- Use pathlib.Path instead of strings and os.path methods in some places
+- Add kernel-only config options 'KernelDebuggerPath' and 'RetraceEnvironment=native'
+- Introduce type annotations
+- Address issues reported by Pylint
+- Other minor code refactoring and cleanup operations
+
```
Comment (mkutlak) [inline retrace-server.spec.in:222]: This should be in `retrace-server.spec` instead.

**Review by mkutlak — DISMISSED** [review]

**Comment by mgrabovsky** [comment]
Ah, finally it makes sense. Thanks a lot!

**Review by xsuchy — APPROVED** [review]

**Comment by xsuchy** [comment]
Can you please save test/Makefile?

**Comment by mgrabovsky** [comment]
I will deal with the test later. ~It can be done with Meson, but I'm struggling with the details right now.~

I managed to write a working Meson script for the test, but the test fails anyway, so I'll try to fix it and introduce the script in a new PR.

Merging this for today.

**Pull request merged**